manifests/install/cni_plugins.pp

```diff
@@ -27,20 +27,30 @@
       $_url = k8s::format_url($download_url_template, {
           version => $version,
       })
+      $_target = "/opt/k8s/cni-${version}";
+      $_tarball_target = '/opt/k8s/archives';
 
-      file { '/opt/cni/bin':
-        ensure => directory,
+      file { $_target:
+        ensure  => stdlib::ensure($ensure, 'directory'),
       }
 
       archive { 'cni-plugins':
         ensure       => $ensure,
-        path         => "/tmp/cni-plugins-linux-${version}.tgz",
+        path         => "${_tarball_target}/cni-plugins-linux-${version}.tgz",
         source       => $_url,
         extract      => true,
-        extract_path => '/opt/cni/bin',
-        creates      => '/opt/cni/bin/bridge',
+        extract_path => $_target,
+        creates      => "${_target}/bridge",
         cleanup      => true,
-        require      => File['/opt/cni/bin'],
+      }
+
+      file { '/opt/cni/bin':
+        ensure  => stdlib::ensure($ensure, 'link'),
+        mode    => '0755',
+        replace => true,
+        force   => true,
+        target  => $_target,
+        require => Archive['cni-plugins'],
```

Comment:
This could likely cause issues on Debian, as it installs `/opt/cni/bin` as a directory, also not sure how SELinux would deal with it.

Reply:
I've had the time to test this on relevant Debian versions now, and it doesn't completely break in the ways I feared it would - but it does come with the caveat that apt will happily replace any extracted cni binaries in case it decides to update the package.

Nothing that blocks the change either way, though it will probably need a few notes on various caveats - like the apt thing, and that updating the CNI binaries will remove any that are provided by cluster services. (e.g. multus)

As a potential option that makes it a little safer, it might be possible to - instead of symlinking the entire folder - individually symlink the CNI plugins into the folder. But that's something that can be done later if it turns out that the full-folder symlink causes issues.